pkg/common/watchdog.go

```diff
@@ -68,3 +68,65 @@ func StartWatchdog(c WatchdogConfig) *WatchdogChannel {
 
 	return &WatchdogChannel{incoming, sync.Mutex{}, false}
 }
+
+type Watchdog struct {
+	// Timeout after which `OnTimeout` is called.
+	timeout time.Duration
+	// A closure that is called once `Timeout` is reached.
+	onTimeout   func()
+	notify      chan struct{}
+	closeSignal chan struct{}
+}
+
+func NewWatchdog(timeout time.Duration, onTimeout func()) *Watchdog {
+	return &Watchdog{timeout: timeout, onTimeout: onTimeout, notify: make(chan struct{}), closeSignal: make(chan struct{})}
+}
+
+func (w Watchdog) Start() chan struct{} {
+	terminate := make(chan struct{})
```

Comment:
Do we need a `terminate` channel here? From what I can see, we don't use (won't use it) and it's only used for the tests.

Reply:
Ok, agree!